File: python/7_kyu/isograms.py
# Isograms
# https://www.codewars.com/kata/54ba84be607a92aa900000f1/train/python

def is_isogram(s):
    letters = []

    for i in list(s.lower()):
        if i not in letters:
            letters.append(i)
        else:
            return False
    
    return True
    
# # Best Solution
# def is_isogram(string):
#     return len(string) == len(set(string.lower()))

print(is_isogram("Dermatoglyphics"))
print(is_isogram("isIsogram"))

File: javascript/7_kyu/isograms.js
// Isograms
// https://www.codewars.com/kata/54ba84be607a92aa900000f1/train/javascript

function isIsogram(str) {
    // create a set out of the string, ensuring that the elements of the set are lowercase
    unique = new Set(str.toLowerCase());

    // revert the set into an array and compare its length to the length of the original string
    return str.length == [...unique].length;
}

console.log(isIsogram("Dermatoglyphics"));